core/storage/user/gae_models.py

```diff
@@ -144,3 +144,53 @@ def get(cls, user_id, exploration_id):
         instance_id = cls._generate_id(user_id, exploration_id)
         return super(ExplorationUserDataModel, cls).get(
             instance_id, strict=False)
+
+
+class ExplorationInCollectionCompletionModel(base_models.BaseModel):
+    """Stores all explorations which have been completed within the context of
+    a collection.
+    """
+
+    # The user id.
+    user_id = ndb.StringProperty(required=True, indexed=True)
+    # The collection id.
+    collection_id = ndb.StringProperty(required=True, indexed=True)
+    # Contains the list of explorations which have been completed within the
+    # context of the collection specified by the above ID.
+    context = ndb.JsonProperty(required=True, indexed=False, default={})
```

Comment:
Weird name. Why 'context'? I mean, this doesn't store the context...

Also, is this just a list of strings? If so, could do ndb.StringProperty(repeated=True, ...). Or are you planning to extend it later?

In any case, comment in line above should be updated to indicate exactly what this is.